src/java/org/apache/cassandra/index/sai/disk/StorageAttachedIndexWriter.java

```diff
@@ -212,9 +208,10 @@ public void abort(Throwable accumulator, boolean fromIndex)
         // Mark the write operation aborted, so we can short-circuit any further operations on the component writers.
         aborted = true;
         
+        // TODO: This really shouldn't be necessary if we fail in the right places?
         // Make any indexes involved in this transaction non-queryable, as they will likely not match the backing table.
-        if (fromIndex)
-            indexes.forEach(StorageAttachedIndex::makeIndexNonQueryable);
+        //if (makeNonQueryable)
+        //    indexes.forEach(StorageAttachedIndex::makeIndexNonQueryable);
```

Comment:
The changes in `abort()` here are probably what require some real discussion. The position I'm going to try to defend here is that making the index non-queryable is sub-optimal in all cases if our goal is to keep indexes and backing tables consistent w/ one another. (It is breaking a couple tests right now, but I'm working on that...)

(I don't see any problem w/ continuing to not throw when the abort is upstream is.)

@adelapena @mike-tr-adamson